### PR TITLE
Placement Application Report Improvements

### PR DESCRIPTION
**Pull request by davidatkinsuk**
Following a review by users of the report in production, the following improvements have been made:

* ‘Submission Date’ has been relabelled as ‘Placement Request Submission Date’ to disambiguate it from the Application Submission Date
* A new column ‘Application Submission Date’ has been added
* We no longer show Placement Applications that have not been submitted

example report generated from the integration test is attached 
[report.xlsx](https://github.com/ministryofjustice/hmpps-approved-premises-api/files/13744017/report.xlsx)

we now have the following columns in the report:

1. crn
2. applicationSubmittedAt
3. applicationAssessedDate
4. assessorCru
5. assessmentDecision
6. assessmentDecisionRationale
7. placementRequestId
8. placementRequestSubmittedAt
9. requestedArrivalDate
10. requestedDurationDays
11. decision
12. decisionMadeAt
13. ageInYears
14. gender
15. mappa
16. offenceId
17. noms
18. sentenceType
19. releaseType
20. referralLdu
21. referralTeam
22. referralRegion
23. referrerUsername
24. targetLocation
25. applicationWithdrawalDate
26. applicationWithdrawalReason
27. bookingID
28. bookingCancellationReason
29. bookingCancellationDate
30. expectedArrivalDate
31. expectedDepartureDate
32. premisesName
33. actualArrivalDate
34. actualDepartureDate
35. departureReason
36. departureMoveOnCategory
37. hasNotArrived
38. notArrivedReason
39. placementRequestType
40. paroleDecisionDate

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/AssertJExtensions.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.util
+
+import org.assertj.core.api.AbstractLocalDateTimeAssert
+import org.assertj.core.api.AbstractOffsetDateTimeAssert
+import org.assertj.core.api.Assertions
+import java.time.LocalDateTime
+import java.time.OffsetDateTime
+import java.time.temporal.ChronoUnit
+
+fun AbstractOffsetDateTimeAssert<*>.isWithinTheLastMinute() {
```
Comment (pezholio) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/util/AssertJExtensions.kt:10]: ⭐

File: src/main/resources/db/migration/all/20231221142033__backfill_placement_application_decision_made.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+UPDATE placement_applications pa SET decision_made_at = (SELECT min(pr.created_at) FROM placement_requests pr WHERE pr.placement_application_id = pa.id)
```
Comment (pezholio) [inline src/main/resources/db/migration/all/20231221142033__backfill_placement_application_decision_made.sql:1]: Thinking we should only set this on placment_applications with a decision i.e:

```
UPDATE placement_applications pa SET decision_made_at = (SELECT min(pr.created_at) FROM placement_requests pr WHERE pr.placement_application_id = pa.id) WHERE decision IS NOT NULL
```

**Review by pezholio — COMMENTED** [review]
LGTM - just one minor concern about the migrations

File: src/main/resources/db/migration/all/20231221142033__backfill_placement_application_decision_made.sql (hunk 2 above)
Comment (davidatkinsuk) [reply src/main/resources/db/migration/all/20231221142033__backfill_placement_application_decision_made.sql]: oh yeah, good call. fixed

**Review by pezholio — APPROVED** [review]
LGTM 🎄

**Pull request merged**